platform/openide.util/test/unit/src/org/openide/util/BaseUtilitiesTest.java

```diff
@@ -557,4 +549,102 @@ public void test_toPrimitiveArray_returnsArrayOfPrimitives_whenGivenArrayOfObjec
+        
+        assertEquals(" \t", pureClassName(" \t"));
+    }
+    //--------------------------------------------------------------------------
+    public void test_pureCalssName__returnsClassName_whenGivenFullyQualifiedClassName() {
+        
+        assertEquals("java.lang.String", pureClassName("java.lang.String"));
+        assertEquals("UnicodeBlock", pureClassName("java.lang.Character$UnicodeBlock"));
+    }
+    //--------------------------------------------------------------------------
+    public void test_wrapString_throwsNullPointer_whenGivenNullArgument() {
+
+        expectNullPointer(() -> wrapString(null, 11, this.breakIterator, true));
+
+        //NullPointerException happens only when line breaking actually happenes.
+        //Change second argument below from 1 to 10 and and rerun the test to observe this.
+        //Maybe we should tighten this behavior to throw NPE unconditionally.
+        expectNullPointer(() -> wrapString("ab\ncd", 1, null, true));
+    }
+    //--------------------------------------------------------------------------
+    public void test_wrapString_assumesWidthEqalsOne_whenGivenZeroWidth() {
+
+        assertEquals("a\n\n \n\nb\n", wrapString("a\nb", 0, this.breakIterator, true));
+    }
+    //--------------------------------------------------------------------------
+    public void test_wrapString_assumesWidthEqalsOne_whenGivenNegativeWidth() {
+
+        assertEquals("a\n\n \n\nb\n", wrapString("a\nb", -1, this.breakIterator, true));
+    }
+    //--------------------------------------------------------------------------
+    public void test_wrapString_returnsblankString_whenGivenEmptyString() {
+
+        assertEquals("\n", wrapString("", 10, this.breakIterator, true));
+        assertEquals(" \n", wrapString(" ", 10, this.breakIterator, true));
+        assertEquals(" \n", wrapString("\n", 10, this.breakIterator, true));
+    }
+    //--------------------------------------------------------------------------
+    public void test_wrapString_returnsOneLineWithOriginalString_whenGivenShortString() {
+
+        // String shorter than given width (10).
+        assertEquals("abc def\n", wrapString("abc\ndef", 10, this.breakIterator, true));
+    }
+    //--------------------------------------------------------------------------
+    public void test_wrapString_wrapsLinesEvenInShostStrings_whenRemoveNewLinesEqualsFalse() {
+
+        // String shorter than given width (10).
+        assertEquals("abc\ndef\n", wrapString("abc\ndef", 10, this.breakIterator, false));
+    }
+    //--------------------------------------------------------------------------
+    public void test_wrapString_wrapsString_whenGivenLongString() {
+
+        assertEquals("abc \ndef\n", wrapString("abc\ndef", 5, this.breakIterator, true));
+    }
+    //--------------------------------------------------------------------------
+    private void expectNullPointer(final Runnable code) {
```

Comment:
Wouldn't a Supplier<T> be more appropriate ?

Reply:
No. Supplier is used to retrieve some value. Here the function just executes a given code block and does not expect it to return any value. Runnable does the job.

Reply:
yes but Runnable while being a functional interface brings in the idea of threads. Maybe be more generic on this.

Reply:
I understand Runnable as something that can be "run", not necessarily by another thread. So now we have unintended "personal preferences conflict" :((( Any ideas how to solve it?